android-app-examples/SNetDemo/app/src/main/java/com/example/snetdemo/ImageSegmentationActivity.java

```diff
@@ -0,0 +1,449 @@
+        }
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.CAMERA}, 1);
+        }
+
+        setTitle("Image Segmentation Demo");
+
+        btn_UploadImageInput = (Button) findViewById(R.id.btn_uploadImageForSegmentation);
+        btn_RunImageSegmentation = (Button) findViewById(R.id.btn_runImageSegmentation);
+        btn_RunImageSegmentation.setEnabled(false);
+
+        btn_GrabCameraImage = (Button) findViewById(R.id.btn_grabCameraImage);
+
+        imv_Input = (ImageView)findViewById(R.id.imageViewInput);
+
+        loadingPanel = (RelativeLayout) findViewById(R.id.loadingPanel);
+        loadingPanel.setVisibility(View.INVISIBLE);
+
+        textViewProgress = findViewById(R.id.textViewProgress);
+        textViewProgress.setVisibility(View.INVISIBLE);
+
+        textViewResponseTime = findViewById(R.id.textViewResponseTime);
+        textViewResponseTime.setText("Service response time (ms):");
+        textViewResponseTime.setVisibility(View.INVISIBLE);
+
+
+        PackageManager pm = this.getPackageManager();
+        if (!pm.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY))
+        {
+            mIsDeviceWithCamera = false;
+            btn_GrabCameraImage.setEnabled(false);
+        }
+
+        channelID = this.getResources().getInteger(R.integer.channel_id);
+
+        AsyncTask task = new AsyncTask<Object, Object, Object>()
+        {
+
+            protected void onPreExecute()
+            {
+                super.onPreExecute();
+
+                textViewProgress.setVisibility(View.VISIBLE);
+                textViewProgress.setText("Opening service channel");
+                loadingPanel.setVisibility(View.VISIBLE);
+
+                getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                        WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+
+
+            }
+
+            @Override
+            protected Object doInBackground(Object... param)
+            {
+                SnetSdk sdk = new SnetSdk(ImageSegmentationActivity.this);
+                semanticSegmentationService = new SemanticSegmentationService(sdk, BigInteger.valueOf(channelID));
+                return null;
+            }
+
+            protected void onPostExecute(Object obj)
+            {
+                textViewProgress.setVisibility(View.INVISIBLE);
+                loadingPanel.setVisibility(View.INVISIBLE);
+                getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+            }
+
+
+        };
+        task.execute();
+
+    }
+
+    public void sendRunImageSegmentationMessage(View view)
+    {
+        if(this.mIsInputImageUploaded)
+        {
+            AsyncTask task = new AsyncTask<Object, Integer, Object>()
+            {
+                protected void onPreExecute()
```

Comment:
I would reuse `onPreExecute()` and `onPostExecute()` for different tasks in activity if it is possible.